src/core/atomicdex/services/mm2/mm2.service.cpp

```diff
@@ -702,11 +740,134 @@ namespace atomic_dex
                                             SPDLOG_WARN("Should set to false the active field in cfg for: {} - reason: {}", tickers[idx], error);
                                         }
                                     }
-                                    idx += 1;
                                     if (res)
                                     {
-                                        this->process_balance_answer(answer);
+                                        if (answer.contains("result"))
+                                        {
+                                            if (answer["result"].contains("task_id"))
+                                            {
+                                                auto task_id = answer.at("result").at("task_id").get<std::int8_t>();
+                                                {
+                                                    using namespace std::chrono_literals;
+
+                                                    static std::size_t z_nb_try      = 0;
+                                                    nlohmann::json     z_error       = nlohmann::json::array();
+                                                    nlohmann::json     z_batch_array = nlohmann::json::array();
+                                                    t_init_z_coin_status_request z_request{.task_id = task_id};
+
+                                                    SPDLOG_DEBUG("{} init_z_coin Task ID: {}", tickers[idx], task_id);
+
+                                                    nlohmann::json j = ::mm2::api::template_request("init_z_coin_status", true);
+                                                    ::mm2::api::to_json(j, z_request);
+                                                    z_batch_array.push_back(j);
+                                                    std::string last_event = "none";
+                                                    std::string event = "none";
+
+                                                    // set to enabled "early" so it shows up in models
```

Comment:
Uncommenting this will show the coin in wallet pages, but blocks the login or enable screen which waiting.
Commenting it out, coin will begin to enable but wont be removed from enable menu or added to wallet pages.

Because `currently_enabled is boolean, we'll need an extra state here like `enabling_in_progress`.